### ISSUE TITLE
web.config and IIS "InProcess"

### ISSUE THREAD

**Issue by chekm8** [aspnet/IISIntegration#1424]
I was attempting to get IIS Express to run an ASP.Net Core Web Application  using the new "InProcess" feature so that I can access ServerVariables in IIS.  From previous tries to get this to work on Core 2.1 I had a web.config file with the following content:

```
  <system.webServer>
    <handlers>
      <remove name="aspNetCore" />
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout">
      <environmentVariables />
    </aspNetCore>
  </system.webServer>
```
After I installed 15.9 preview of visual studio it gave us the tooling to change the "InProcess" flag from the csproj properties.  When I did this and reran my project it would rewrite the web.config like this:
```
  <system.webServer>
    <handlers>
      <remove name="aspNetCore" />
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="InProcess">
      <environmentVariables>
        <environmentVariable name="ASPNETCORE_HTTPS_PORT" value="44380" />
        <environmentVariable name="ASPNETCORE_ENVIRONMENT" value="Development" />
      </environmentVariables>
    </aspNetCore>
  </system.webServer>

```
A side effect of this is that although the project would run, it no longer hits any break points.  

If I comment out the config section completely or turn off "InProcess" on the csproj properties the debugger would begin working again.

**Repro steps:**
**Using VS 15.9 preview:**

1. Create an asp core web application, use MVC
2. Add a web.config using the add new item menu.
3. uncomment the system.webserver config that is supplied by default
4. Change the csproj properties => debug => "InProcess" 
5. put a break point on the homecontroller's index method and run the project using iisexpress
6. break point isn't hit.

**Comment by jkotalik** [comment]
```<system.webServer>
    <handlers>
      <remove name="aspNetCore" />
      <add name="aspNetCore" path="*" verb="*" modules="AspNetCoreModule" resourceType="Unspecified" />
    </handlers>
    <aspNetCore processPath="%LAUNCHER_PATH%" arguments="%LAUNCHER_ARGS%" stdoutLogEnabled="false" stdoutLogFile=".\logs\stdout" hostingModel="InProcess">
      <environmentVariables>
        <environmentVariable name="ASPNETCORE_HTTPS_PORT" value="44380" />
        <environmentVariable name="ASPNETCORE_ENVIRONMENT" value="Development" />
      </environmentVariables>
    </aspNetCore>
  </system.webServer>
```

To confirm, modules="AspNetCoreModule" should be modules="AspNetCoreModuleV2". Can you confirm that is the case?

**Comment by jkotalik** [comment]
CC @BillHiebert seems like this is an issue on your side.

**Comment by jkotalik** [comment]
1. If there is a web.config file already present, are we allowed to modify  it at all? 
2. It seems that the new web.config file isn't changing the modules section to`modules="AspNetCoreModuleV2"`

**Comment by jkotalik** [comment]
Fix will be in VS. Leaving issue open for tracking. 

Thanks @chekm8 for finding this issue.

**Comment by chekm8** [comment]
> Fix will be in VS. Leaving issue open for tracking.
> Thanks @chekm8 for finding this issue.

@jkotalik I tried to change my web.config to use "AspNetCoreModuleV2" to verify if it works, but when I rebuild and run the app it automatically rewrites the web.config without the value.  Is there anyway to stop the autorewrite so I can verify?

**Comment by chekm8** [comment]
ok, but I should be able to have my own web.config in the app after the fix?  I'm just thinking it might be necessary for something else that may be needed.  
Asking out of curiosity, since I'm used to running full framework apps.

**Comment by jkotalik** [comment]
Yes, you are still allowed to have your own web.config.

**Comment by chekm8** [comment]
Thanks

**Comment by jkotalik** [comment]
Will be fixed in 15.9-preview4
